webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -0,0 +1,66 @@
+import {
+  NOTIFICATIONS_GET_NOTIFICATIONS,
+  NOTIFICATIONS_TOGGLE_DRAWER,
+  NOTIFICATIONS_SET_EXPANDED_GROUP,
+  NOTIFICATIONS_MARK_AS_READ
+} from '../../consts';
+import { notificationsDrawer } from '../../../common/sessionStorage';
+import API from '../../../API';
+const getNotificationsInterval = 10000;
+
+export const getNotifications = url => dispatch => {
+  if (
+    document.visibilityState === 'visible' ||
+    document.visibilityState === 'prerender'
+  ) {
+    API.get(url).then(response => {
+      dispatch({
+        type: NOTIFICATIONS_GET_NOTIFICATIONS,
+        payload: {
+          notifications: response.notifications
+        }
+      });
+      setTimeout(
```

Comment:
right, but in the case where it was skipped (as the tab was not in focus) it should trigger the same check again, the current code will stop first time the condition is not met.